test/functional/feature_block_reward_reallocation.py

```diff
@@ -53,20 +58,27 @@ def create_test_block(self, version=536870912):
         block.solve()
         return block
 
+    def mine_blocks(self, num_blocks, expected_realloc=None):
+        batch = 15  # generate at most 10 blocks at once
+        for i in range((num_blocks - 1) // batch):
+            self.bump_mocktime(batch, False)
```

Comment:
Look at the bump_moctime method. It basically says "update mocktime variable, but don't push that to the nodes", which results in time spent in bump_mocktime from 5 seconds -> 0.01 seconds.

Reply:
>which results in time spent in bump_mocktime from 5 seconds -> 0.01 seconds.

oh, wow.. in total I guess? cause `setmocktime` is a  pretty trivial rpc command, the only narrow place is `cs_main` lock as far as I can tell.

Reply:
yes, 5 seconds was the cumtime spent in bump_mocktime (it was called hundreds / maybe thousands of times) over the whole script before these optimizations, after down to 0.05 (maybe mildly more since I had to introduce a bump_mocktime w/o false down below, but since it's only 1 call, should be basically nothing)

I used `python -m cProfile` sorted by cumtime to find this slowdown